src/main/java/com/czertainly/core/validation/certificate/X509CertificateValidator.java

```diff
@@ -97,14 +105,18 @@ private Map<CertificateValidationCheck, CertificateValidationCheckDto> validateP
         return validationOutput;
     }
 
-    private CertificateValidationCheckDto checkCertificateChain(X509Certificate certificate, X509Certificate issuerCertificate, CertificateValidationStatus issuerCertificateStatus, boolean isCompleteChain) {
+    private CertificateValidationCheckDto checkCertificateChain(X509Certificate certificate, X509Certificate issuerCertificate, boolean trustedCa, CertificateValidationStatus issuerCertificateStatus, boolean isCompleteChain) {
         if (issuerCertificate == null) {
             // should be trust anchor (Root CA certificate)
-            if (isCompleteChain) {
+            if (isCompleteChain && trustedCa) {

```

Comment:
Rewrite if to be more readable, first should be if `isCompleteChain` and inside if `trustedCa` or not.